pkg/model/mask.go

```diff
@@ -135,3 +160,43 @@ func UpdateService(src, dst *Service, mask Mask) error {
 
 	return nil
 }
+
+// UpdateTranslation updates the destination translation based on the source translation and field mask.
+func UpdateTranslation(src, dst *Translation, mask Mask) error {
+	// nonMsgField will store the field paths that are not related to "Messages."
+	nonMsgField := []string{}
+
+	for _, path := range mask {
+		switch path {
+		case "Messages":
```

Comment:
This works only for Translation.

How can you implement this as a general purpose solution in `updateField`? Based on https://github.com/translate-agent/translate/pull/219/files#r1437547444

`when the update mask is a repeated field, we append by default. However, if the repeated message contains the field id (unique identifier), we replace it`